client-v2/src/main/java/com/clickhouse/client/api/Client.java

```diff
@@ -1040,7 +1040,7 @@ public synchronized void register(Class<?> clazz, TableSchema schema) {
         boolean defaultsSupport = schema.hasDefaults();
         tableSchemaHasDefaults.put(schemaKey, defaultsSupport);
         for (ClickHouseColumn column : schema.getColumns()) {
-            String propertyName = column.getColumnName().toLowerCase().replace("_", "").replace(".", "");
```

Comment:
the purpose of this replacement is to leave only words without delimiting characters for easier compare. 
For example, column name `first_name` should match getter method name `firstName`, that is why `_` is removed. 
If you have getters with underscores then is should be replaced from getter method name, too. 
So current fix breaks canonical case when getters are without `_`.